worker/convert/converter.py

```diff
@@ -1,10 +1,39 @@
 import logging
 
 from PIL import Image
+from wand.image import Image as Img
+import pdftotext
 
 
 def convert_tif_to_jpg(source_file, target_file):
     if source_file != target_file:
         logging.getLogger(__name__).debug(f"Converting {source_file} "
                                           f"to {target_file}")
         Image.open(source_file).save(target_file)
+
+
+def convert_pdf_to_txts(source_file, output_dir):
+    logging.getLogger(__name__).debug(f"Creating txt files from {source_file} "
+                                      f"to {output_dir}")
+    with open(source_file, "rb") as input_stream:
+        pdf = pdftotext.PDF(input_stream)
+        index = 0
+        for page in pdf:
+            try:
+                output = open('%s%d.txt' % (output_dir, index), 'wb')
```

Comment:
Mach hier doch aus dem Index ein pdf.index(page). Dann sparst du dir diese gefakete doppelte for schleife. Siehe [diesen Beitrag](https://stackoverflow.com/questions/176918/finding-the-index-of-an-item-given-a-list-containing-it-in-python)

Reply:
pdf ist leider keine normale liste sondern ein c object was keine .index() methode zur verfügung stellt.